Error `over` array containing strings

```
func TestKazaamOverArrayStrings(t *testing.T) {
        spec := `[{
                "operation": "shift",
                "over": "doc.guidObjects",
                "spec": {"raw": "$"}
        }]`
        jsonIn := `{"doc":{"guidObjects":["foo",5,false]}}`
        jsonOut := `{"doc":{"guidObjects":[{"raw":"foo"},{"raw":5},{"raw":false}]}}`

        kazaamTransform, _ := kazaam.NewKazaam(spec)
        kazaamOut, _ := kazaamTransform.TransformJSONStringToString(jsonIn)
        areEqual, _ := checkJSONStringsEqual(kazaamOut, jsonOut)

        if !areEqual {
                t.Error("Transformed data does not match expectation.")
                t.Log("Expected: ", jsonOut)
                t.Log("Actual:   ", kazaamOut)
                t.FailNow()
        }
}
```

```
--- FAIL: TestKazaamOverArrayStrings (0.00s)
        kazaam_int_test.go:453: Transformed data does not match expectation.
        kazaam_int_test.go:454: Expected:  {"doc":{"guidObjects":[{"raw":"foo"},{"raw":5},{"raw":false}]}}
        kazaam_int_test.go:455: Actual:    {"doc":{"guidObjects":[{"raw":foo},{"raw":5},{"raw":false}]}}
```

--- a/glide.yaml
+++ b/glide.yaml
@@ -1,6 +1,6 @@
 package: github.com/qntfy/kazaam
 import:
 - package: github.com/qntfy/jsonparser
-  version: ^1.0.1
+  version: ^1.0.2
 - package: github.com/satori/go.uuid
   version: v1.1.0

--- a/kazaam.go
+++ b/kazaam.go
@@ -197,7 +197,7 @@ func (k *Kazaam) TransformInPlace(data []byte) ([]byte, error) {
 		if specObj.Config != nil && specObj.Over != nil {
 			var transformedDataList [][]byte
 			_, err = jsonparser.ArrayEach(data, func(value []byte, dataType jsonparser.ValueType, offset int, err error) {
-				transformedDataList = append(transformedDataList, value)
+				transformedDataList = append(transformedDataList, transform.HandleUnquotedStrings(value, dataType))
 			}, strings.Split(*specObj.Over, ".")...)
 			if err != nil {
 				return data, transformErrorType(err)

--- a/kazaam_int_test.go
+++ b/kazaam_int_test.go
@@ -435,3 +435,24 @@ func TestKazaamTransformDelete(t *testing.T) {
 		t.FailNow()
 	}
 }
+
+func TestKazaamOverArrayStrings(t *testing.T) {
+	spec := `[{
+                "operation": "shift",
+                "over": "doc.guidObjects",
+                "spec": {"raw": "$"}
+        }]`
+        jsonIn := `{"doc":{"guidObjects":["foo",5,false]}}`
+        jsonOut := `{"doc":{"guidObjects":[{"raw":"foo"},{"raw":5},{"raw":false}]}}`
+
+        kazaamTransform, _ := kazaam.NewKazaam(spec)
+        kazaamOut, _ := kazaamTransform.TransformJSONStringToString(jsonIn)
+        areEqual, _ := checkJSONStringsEqual(kazaamOut, jsonOut)
+
+        if !areEqual {
+                t.Error("Transformed data does not match expectation.")
+                t.Log("Expected: ", jsonOut)
+                t.Log("Actual:   ", kazaamOut)
+                t.FailNow()
+        }
+}

--- a/transform/util.go
+++ b/transform/util.go
@@ -69,7 +69,7 @@ func getJSONRaw(data []byte, path string, pathRequired bool) ([]byte, error) {
 
 				// use jsonparser.ArrayEach to copy the array into results
 				_, err := jsonparser.ArrayEach(data, func(value []byte, dataType jsonparser.ValueType, offset int, err error) {
-					results = append(results, handleUnquotedStrings(value, dataType))
+					results = append(results, HandleUnquotedStrings(value, dataType))
 				}, beforePath...)
 				if err == jsonparser.KeyPathNotFoundError {
 					if pathRequired {
@@ -121,7 +121,7 @@ func getJSONRaw(data []byte, path string, pathRequired bool) ([]byte, error) {
 	if dataType == jsonparser.String {
 		// bookend() is destructive to underlying slice, need to copy.
 		// extra capacity saves an allocation and copy during bookend.
-		result = handleUnquotedStrings(result, dataType)
+		result = HandleUnquotedStrings(result, dataType)
 	}
 	if len(result) == 0 {
 		result = []byte("null")
@@ -283,7 +283,7 @@ func bookend(value []byte, bef, aft byte) []byte {
 }
 
 // jsonparser strips quotes from returned strings, this adds them back
-func handleUnquotedStrings(value []byte, dt jsonparser.ValueType) []byte {
+func HandleUnquotedStrings(value []byte, dt jsonparser.ValueType) []byte {
 	if dt == jsonparser.String {
 		// bookend() is destructive to underlying slice, need to copy.
 		// extra capacity saves an allocation and copy during bookend.

